src/macaron/dependency_analyzer/repo_finder.py

```diff
@@ -0,0 +1,185 @@
+        The HTTP session to use for attempting the GET request.
+    url : str
+        The URL for the GET request.
+
+    Returns
+    -------
+    str :
+        The retrieved file data or an empty string.
+    """
+    if not url.endswith(".pom"):
+        return ""
+    res = session.get(url)
+    if not res.ok:
+        logger.warning("Failed to retrieve pom from: %s, error code: %s", url, res.status_code)
+        return ""
+    logger.info("Found artifact POM at: %s", url)
+    return res.text
+
+
+def _find_element(parent: typing.Any, target: str) -> typing.Any:
+    # Attempt to match the target tag within the children of parent.
+    for child in parent:
+        # Account for raw tags and tags accompanied by Maven metadata enclosed in curly braces. E.g. '{metadata}tag'
+        if child.tag == target or child.tag.endswith("}" + target):
+            return child
+    return None
+
+
+def parse_pom(pom: str, tags: list[str]) -> list[str]:
+    """
+    Parse the passed pom and extract the passed tags.
+
+    Parameters
+    ----------
+    pom : str
+        The POM as a string
+    tags : list[str]
+        The list of tags to try extracting from the POM
+
+    Returns
+    -------
+    list[str] :
+        The extracted contents of any matches tags
+    """
+    xml = fromstring(pom)
+    results = []
+
+    # Try to match each tag with the contents of the POM
+    for tag in tags:
+        element = xml
+        tag_parts = tag.split(".")
+        for index, tag_part in enumerate(tag_parts):
+            element = _find_element(element, tag_part)
+            if element is None:
+                break
+            if index == len(tag_parts) - 1:
+                # Add the contents of the final tag
+                results.append(element.text)
+
+    return results
+
+
+def find_repo(gav: str, tags: list[str]) -> list[str]:
+    """
+    Attempt to retrieve a repository URL that matches the passed GAV artifact.
+
+    Parameters
+    ----------
+    gav : str
+        An artifact represented as a GAV (group, artifact, version) in the format: G:A:V.
+    tags : list[str]
+        The list of XML tags to look for, each in the format: tag1[.tag2 ... .tagN]
+
+    Returns
+    -------
+    list[str] :
+        The URLs found for the passed GAV.
+    """
+    if len(global_config.artifact_repositories) == 0:
+        logger.warning("No repositories set in config artifact_repositories parameter for repo finder.")
+        return []
+
+    # Parse the GAV and create the URLs for its POM
+    request_urls = parse_gav(gav)
+    if len(request_urls) == 0:
+        return []
+
+    # Try each POM URL in order, terminating early if a match is found
+    session = requests.Session()
+    pom = ""
+    for request_url in request_urls:
+        pom = retrieve_pom(session, request_url)
+        if pom != "":
+            break
+
+    if pom == "":
+        return []
+
+    # Parse XML data and return URL
+    return parse_pom(pom, tags)
```

Comment:
This `parse_pom` does not remove the trailing spaces or new lines. So if a tag in the xml file is 
```
<url>
         https://example.example/example
</url>
```
The result would be something similar to 
```
['\n         https://example.example/example\n        ']
```
In this case, I think the returned URL might be incorrect.

Reply:
Good catch. I'll fix that.